diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 27d0e3a0073c20adff764c71575e200d6ac23432..0000000000000000000000000000000000000000
diff --git a/flash/README.md b/flash/README.md
deleted file mode 100644
index 1576ebf1a99dab1462caa81c14240864ed1454cb..0000000000000000000000000000000000000000
diff --git a/flash/package.json b/flash/package.json
deleted file mode 100644
index 92689fa9b24e93a631d2b52572030d7e17ccb10f..0000000000000000000000000000000000000000
diff --git a/lib/asn1.js b/lib/asn1.js
index 4025f8a9e333c3c7b0547f9445f69ed818c86500..37d228367da3a413f81b54df2588fbdce75f8d56 100644
--- a/lib/asn1.js
+++ b/lib/asn1.js
@@ -448,6 +448,9 @@ asn1.fromDer = function(bytes, options) {
   if(!('decodeBitStrings' in options)) {
     options.decodeBitStrings = true;
   }
+  if(!('maxDepth' in options)) {
+    options.maxDepth = 200;
+  }
 
   // wrap in buffer if needed
   if(typeof bytes === 'string') {
@@ -476,6 +479,13 @@ asn1.fromDer = function(bytes, options) {
  * @return the parsed asn1 object.
  */
 function _fromDer(bytes, remaining, depth, options) {
+  // MAX_ASN1_DEPTH_GUARD
+  if(typeof options.maxDepth === "number" && depth > options.maxDepth) {
+    var error = new Error("ASN.1 input exceeds maximum depth.");
+    error.depth = depth;
+    error.maxDepth = options.maxDepth;
+    throw error;
+  }
   // temporary storage for consumption calculations
   var start;
 
